src/fetchers.ts

```diff
@@ -0,0 +1,141 @@
+import { PluginMeta } from '@grafana/data';
+import { config, getBackendSrv } from '@grafana/runtime';
+import { importAllDashboards } from 'dashboards/loader';
+import { HostedInstance, RegistrationInfo } from 'types';
+import { createDatasource, findHostedInstance, getHostedLokiAndPrometheusInfo } from 'utils';
+
+export const fetchPluginSettings = async (pluginId: string) =>
+  await getBackendSrv().get(`api/plugins/${pluginId}/settings`);
+
+export const updatePluginJsonData = async (pluginId: string, jsonData: any) => {
+  const pluginSettings = await fetchPluginSettings(pluginId);
+  await getBackendSrv().post(`api/plugins/${pluginId}/settings`, {
+    ...pluginSettings,
+    jsonData: {
+      ...(pluginSettings.jsonData ?? {}),
+      ...jsonData,
+    },
+  });
+};
```

Comment:
**Nitpick**
Hmm... maybe it's just me, but I probably wouldn't look for POST / PUT / DELETE methods under a module called `fetchers`.